internal/entitlement/balanceworker/worker.go

```diff
@@ -169,78 +169,88 @@ func (w *Worker) Close() error {
 func (w *Worker) handleEvent(msg *message.Message) ([]*message.Message, error) {
 	w.opts.Logger.Debug("received system event", w.messageToLogFields(msg)...)
 
-	ceType, found := msg.Metadata[publisher.CloudEventsHeaderType]
+	ceType, found := msg.Metadata[marshaler.CloudEventsHeaderType]
 	if !found {
 		w.opts.Logger.Warn("missing CloudEvents type, ignoring message")
 		return nil, nil
 	}
 
 	switch ceType {
 	// Entitlement events
-	case entitlement.EntitlementCreatedEvent{}.Spec().Type():
-		event, err := spec.ParseCloudEventFromBytes[entitlement.EntitlementCreatedEvent](msg.Payload)
-		if err != nil {
+	case entitlement.EntitlementCreatedEvent{}.EventName():
```

Comment:
The subscriber/consumer part is intentionally left the old way, as I don't want to further complicate this PR, it will be a separate one.